src/tools/rbd/action/Children.cc

```diff
@@ -16,26 +16,62 @@ namespace children {
 namespace at = argument_types;
 namespace po = boost::program_options;
 
-int do_list_children(librbd::Image &image, Formatter *f)
+int do_list_children(librados::IoCtx &io_ctx, librbd::Image &image,
+                     bool all_flag, Formatter *f)
 {
-  std::set<std::pair<std::string, std::string> > children;
   int r;
-
-  r = image.list_children(&children);
-  if (r < 0)
-    return r;
-
-  if (f)
-    f->open_array_section("children");
-
-  for (auto &child_it : children) {
-    if (f) {
-      f->open_object_section("child");
-      f->dump_string("pool", child_it.first);
-      f->dump_string("image", child_it.second);
-      f->close_section();
-    } else {
-      std::cout << child_it.first << "/" << child_it.second << std::endl;
+  if (all_flag) {
```

Comment:
Nit: instead of duplicating all this logic, perhaps always use the new ```list_children2``` API and if ```--all``` wasn't specified, don't print any children that are in the trash.